designs/2019-suggestions/README.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2019-07-04
+- RFC PR: https://github.com/eslint/rfcs/pull/30
+- Authors: Ilya Volodin, Dan Abramov
+
+# Code Suggestions
+
+## Summary
+
+This RTC proposes introducing a new mechanism for providing users with suggestions. It's very similar to fixes mechanism that already exists in ESLint, but will provide ability to create multiple suggestions per rule, suggestions will not be automatically applied by using `--fix` flag, and will require user interaction. This feature will only be exposed through the ESLint API, and will not be available on the CLI. 
+
+## Motivation
+
+This has been suggested multiple times before, but the main driving force behind this proposal is ability to provide user with options on how to automatically refactor/fix their code, without a possibility of "stealthily" breaking it. Examples of such suggestions include ability to suggest adding `import` statement if certain code is added to a file by the user, ability to provide refactoring suggestions like extracting repeated value into a variable, reformatting code to extract repeated statements into reusable function, etc. I do not envision ESLint team to initially create any rules that would provide suggestions, so this would be a feature that would enable plugin authors to do so, but it might be something that ESLint team will eventually do. This feature is geared towards code editor integrations, as such, it will only be available via NodeJS API.
+
+## Detailed Design
+
+* Modify `context.report` function parameter to include new property called `suggest` of type array of functions. Each function will be able to modify the code of the file to implement suggestion. Syntax is going to be exactly the same as for `fix` functions. Example of new signature below:
+```js
+context.report({
+    node,
+    loc,
+    messageId: 'messageId',
+    fix(fixer) { },
+    suggest: [
+        (fixer) => { },
+        (fixer) => { }
+    ]
+});
+```
+* Change `_verifyWithoutProcessors` and `_verifyWithProcessor` functions in linter.js to modify output of those functions to include `suggestions` array returned by the rule.
```

Comment:
I'm not sure that's needed. `disableFixes` was added to avoid auto-fixing code that might result in errors due to unreliable location data. Since suggestions will not be applied automatically, integrations can just ignore suggestions without the need to disable them.

Reply:
I think using `options.disableFixes` is fine. If processor has unreliable location data for fixes, it will also have unreliable location for suggestions as well.